### PR TITLE
Add ARM Managed Application (aka Appliances) spec

### PR DESCRIPTION
**Pull request by vivsriaus**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: arm-resources/appliances/2016-09-01-preview/swagger/appliances.json [hunk 1]
```diff
@@ -0,0 +1,1262 @@
...
+            "description": "The name of the appliance.",
+            "minLength": 3,
+            "maxLength": 64
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/Appliance"
```
Comment (jianghaolu) [inline arm-resources/appliances/2016-09-01-preview/swagger/appliances.json]: a PATCH operation's request body cannot have required properties.

File: arm-resources/appliances/2016-09-01-preview/swagger/appliances.json [hunk 2]
```diff
@@ -0,0 +1,1262 @@
...
+            "type": "string",
+            "description": "The fully qualified ID of the appliance, including the appliance name and the appliance resource type. Use the format, /subscriptions/{guid}/resourceGroups/{resource-group-name}/Microsoft.Solutions/appliances/{appliance-name}",
+            "x-ms-skip-url-encoding": true
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/Appliance"
```
Comment (jianghaolu) [inline arm-resources/appliances/2016-09-01-preview/swagger/appliances.json]: a PATCH operation's request body cannot have required properties.

File: arm-resources/appliances/2016-09-01-preview/swagger/appliances.json [hunk 3]
```diff
@@ -0,0 +1,1262 @@
...
+          }
+        },
+        "x-ms-long-running-operation": true
+      }
+    }
+  },
+  "definitions": {
+    "Appliance": {
+      "properties": {
+        "properties": {
```
Comment (jianghaolu) [inline arm-resources/appliances/2016-09-01-preview/swagger/appliances.json]: Consider using x-ms-client-flatten to provide a better end user experience

File: arm-resources/appliances/2016-09-01-preview/swagger/appliances.json [hunk 4]
```diff
@@ -0,0 +1,1262 @@
...
+      "allOf": [
+        {
+          "$ref": "#/definitions/GenericResource"
+        }
+      ],
+      "description": "Information about appliance."
+    },
+    "ApplianceDefinition": {
+      "properties": {
+        "properties": {
```
Comment (jianghaolu) [inline arm-resources/appliances/2016-09-01-preview/swagger/appliances.json]: Consider using x-ms-client-flatten to provide a better end user experience

File: arm-resources/appliances/2016-09-01-preview/swagger/appliances.json [hunk 5]
```diff
@@ -0,0 +1,1262 @@
...
+          ],
+          "x-ms-enum": {
+            "name": "ResourceIdentityType",
+            "modelAsString": false
+          }
+        }
+      },
+      "description": "Identity for the resource."
+    },
+    "Resource": {
```
Comment (jianghaolu) [inline arm-resources/appliances/2016-09-01-preview/swagger/appliances.json]: Please add a class level description.

File: arm-resources/appliances/2016-09-01-preview/swagger/appliances.json [hunk 6]
```diff
@@ -0,0 +1,1262 @@
...
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
```
Comment (jianghaolu) [inline arm-resources/appliances/2016-09-01-preview/swagger/appliances.json]: Please add `x-ms-examples` to all the APIs: https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/x-ms-examples.md.

**Review by jianghaolu — CHANGES_REQUESTED** [review]

File: arm-resources/appliances/2016-09-01-preview/swagger/appliances.json (hunk 5 above)
Comment (vivsriaus) [inline arm-resources/appliances/2016-09-01-preview/swagger/appliances.json]: Done

File: arm-resources/appliances/2016-09-01-preview/swagger/appliances.json (hunk 4 above)
Comment (vivsriaus) [inline arm-resources/appliances/2016-09-01-preview/swagger/appliances.json]: Thanks. Done

File: arm-resources/appliances/2016-09-01-preview/swagger/appliances.json (hunk 3 above)
Comment (vivsriaus) [inline arm-resources/appliances/2016-09-01-preview/swagger/appliances.json]: Makes sense

File: arm-resources/appliances/2016-09-01-preview/swagger/appliances.json (hunk 2 above)
Comment (vivsriaus) [inline arm-resources/appliances/2016-09-01-preview/swagger/appliances.json]: Thanks!

File: arm-resources/appliances/2016-09-01-preview/swagger/appliances.json (hunk 1 above)
Comment (vivsriaus) [inline arm-resources/appliances/2016-09-01-preview/swagger/appliances.json]: Fixed

File: arm-resources/managedapplications/2016-09-01-preview/swagger/managedapplications.json [hunk 7]
```diff
@@ -225,7 +225,6 @@
           {
             "name": "parameters",
             "in": "body",
-            "required": true,
```
Comment (jianghaolu) [inline arm-resources/managedapplications/2016-09-01-preview/swagger/managedapplications.json]: The properties in the body should also be all optional.
Imagine someone trying to patch an `Appliance`, with this API, that person has to provide `properties` and `kind`, which shouldn't be the case.
Many RPs find adding a `BlahPatchable` class helpful.

**Review by jianghaolu — CHANGES_REQUESTED** [review]

File: arm-resources/managedapplications/2016-09-01-preview/swagger/managedapplications.json (hunk 7 above)
Comment (vivsriaus) [inline arm-resources/managedapplications/2016-09-01-preview/swagger/managedapplications.json]: Makes sense. Fixed

**Review by jianghaolu — APPROVED** [review]

**Comment by jianghaolu** [comment]
Good work! Please let me know when you are done adding examples!

**Comment by vivsriaus** [comment]
@jianghaolu Done adding examples

**Comment by jianghaolu** [comment]
Thanks! Merging.

**Pull request merged**